### PR TITLE
 Add Mxbai(mixbread-ai) Embedding Model to Hub

### PR DESCRIPTION
**Pull request by laxmareddyp**
## Description of the change

Add 4 [mixedbread-ai (mxbai) ](https://huggingface.co/collections/mixedbread-ai/eming-series)embedding model presets  to KerasHub.

License: Apache-2.0 

### Presets

| Preset | Architecture | Pooling | Normalization | Params |
|--------|-------------|---------|---------------|--------|
| `mxbai_embed_large_v1_en` | BERT (24L) | CLS | None | 335M |
| `mxbai_embed_2d_large_v1_en` | BERT (24L) | CLS | None | 335M |
| `mxbai_embed_xsmall_v1_en` | BERT (6L) | Mean | None | 24M |

### Changes

- **`tools/checkpoint_conversion/convert_mxbai_embed_checkpoints.py`** :  Added conversion script for all 4 presets to use `from_preset("hf://...")` to leverage existing BERT/XLM-RoBERTa converters. 
- **`keras_hub/src/utils/transformers/convert_bert.py`**: Bug fix — added `max_sequence_length` to `convert_backbone_config()`. Without this, BERT models with non-512 `max_position_embeddings` (e.g., xsmall-v1 with 4096) fail during weight conversion due to position embedding shape mismatch.

Numeric verification screenshot:

Screenshot 2026-07-27 at 9 44 44 PM

## Checklist

- [x] I have added all the necessary unit tests for my change.
- [x] I have verified that my change does not break existing code and works with all backends (TensorFlow, JAX, and PyTorch).
- [x] My PR is based on the latest changes of the main branch (if unsure, rebase the code).
- [x] I have followed the Keras Hub [Model contribution guidelines](https://github.com/keras-team/keras-hub/blob/master/CONTRIBUTING_MODELS.md) in making these changes.
- [x] I have followed the Keras Hub [API design guidelines](https://github.com/keras-team/keras-hub/blob/master/API_DESIGN_GUIDE.md) in making these changes.
- [x] I have signed the [Contributor License Agreement](https://cla.developers.google.com/about).

### CODE REVIEW

File: tools/checkpoint_conversion/convert_mxbai_embed_checkpoints.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+    hf_search_sims = hf_q @ hf_d.T
+    hf_best = int(np.argmax(hf_search_sims))
+
+    print(
+        f"\nKerasHub sims: {keras_search_sims[0]} -> "
+        f"Best: {documents[keras_best]}"
+    )
+    print(f"HF sims:       {hf_search_sims[0]} -> Best: {documents[hf_best]}")
+    search_match = keras_best == hf_best
+    print(f"Search ranking match: {search_match}")
```
[the remark this answers is not in the document]
Comment (laxmareddyp) [reply tools/checkpoint_conversion/convert_mxbai_embed_checkpoints.py:182]: Ignore this comment from gemini, we are validating the mean logits diff tolerence not max absolute difference.

**Review by divyashreepathihalli — APPROVED** [review]
